libzkbob-rs/src/client/state.rs

```diff
@@ -175,6 +175,40 @@ where
         .collect()
     }
 
+    pub fn get_account(&self, index: u64) -> Option<Account<P::Fr>> {
+        match self.txs.get(index) {
+            Some(Transaction::Account(acc)) => Some(acc),
+            _ => None,
+        }
+    }
+
+    pub fn get_previous_account(&self, index: u64) -> Option<(u64, Account<P::Fr>)> {
+        if index == 0 { return None }
+
+        let prev_acc_indexes = self.txs
+            .iter_slice(0..=(index-1))
+            .filter_map(|(idx, tx)| match tx {
+                Transaction::Account(_) => Some(idx),
+                _ => None,
+            })
+            .max();
```

Comment:
Maybe it will look better this way:
```
self.txs
  .iter_slice(0..=(index-1))
  .filter_map(|(idx, tx)| match tx {
      Transaction::Account(acc) => Some((idx, acc)),
      _ => None,
  })
  .last()
```

Reply:
Do not agree. The SparseArray object `txs` in the common case doesn't guarantee the index order. So the last element isn't 'account with the greatest index'. We are fill it in the serial manner on the top-level. But at this level we cannot make such assumptions

Reply:
Yep, it makes sense.